libs/angular-generator/src/lib/webcomponents-theming.ts

```diff
@@ -0,0 +1,103 @@
+import { ExportSpecifierType, GeneratedFile } from '@ui5/webcomponents-wrapper';
+import { format } from 'prettier';
+import { AngularExportSpecifierType } from './angular-export-specifier-type';
+
+export class WebcomponentsThemingGenerator {
+  public module: WebcomponentsThemingModule;
+  public service: WebcomponentsThemingService;
+
+  constructor() {
+    this.module = new WebcomponentsThemingModule();
+    this.service = new WebcomponentsThemingService();
+  }
+}
+
+export class WebcomponentsThemingModule extends GeneratedFile<AngularExportSpecifierType>{
+  constructor() {
+    super();
+    this.move('main/ui5-webcomponents-theming.module.ts');
+    this.addExport({
+      local: 'Ui5WebcomponentsThemingModule',
+      exported: 'Ui5WebcomponentsThemingModule',
+      types: [ExportSpecifierType.Class, AngularExportSpecifierType.NgModule]
+    })
+  }
+
+  override getCode(): string {
+    this.addImport(
+      ['NgModule', 'ModuleWithProviders', 'Optional', 'isDevMode'],
+      '@angular/core'
+    );
+    const imports: string[] = [];
+    this.addImport(
+      ['Ui5WebcomponentsThemingService'],
+      './ui5-webcomponents-theming.service'
+    );
+
+    return format(
+      [
+        this.getImportsCode(),
+        `
+      const declarations = [${imports.join(', ')}];
+      @NgModule({
+        exports: [],
+        declarations: [],
+        providers: [Ui5WebcomponentsThemingService]
+      })
+      export class Ui5WebcomponentsThemingModule {
+        constructor(public theming: Ui5WebcomponentsThemingService) {
+        }
+      }
+      `,
+      ].join('\n'),
+      { parser: 'typescript' }
+    );
+  }
+}
+
+export class WebcomponentsThemingService extends GeneratedFile {
+  constructor() {
+    super();
+    this.move('main/ui5-webcomponents-theming.service.ts');
+    this.addExport({
+      local: 'Ui5WebcomponentsThemingService',
+      exported: 'Ui5WebcomponentsThemingService',
+      path: 'main/ui5-webcomponents-theming.service.ts',
+      types: [ExportSpecifierType.Class]
+    });
+  }
+
+  override getCode(): string {
+    this.addImport(['Injectable', 'OnDestroy', 'Optional'], '@angular/core');
+    this.addImport(['Ui5ThemingProvider', 'Ui5ThemingService', 'AvailableThemes'], '@ui5/theming-ngx');
```

Comment:
Do we have theming-ngx package as well?

Reply:
Yes. Its a global package that actually controls theming across multiple libraries that are not connected with each other